### PR TITLE
Nodejs: Anchor related changes

### LINKED ISSUE #1458: Nodejs: Anchor related changes
- [x] Add Anchor Output
- [x] Modify Account Output
- [x] Add Anchor Address/ID
- [x] Add Anchor Unlock
- [x] Fix examples
- [x] Fix tests

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes https://github.com/iotaledger/iota-sdk/issues/1458

### CODE REVIEW

**Review by Alex6323 — DISMISSED** [review]

File: bindings/nodejs/lib/types/block/output/output.ts [hunk 1]
```diff
@@ -215,6 +224,66 @@ class AccountOutput extends ImmutableFeaturesOutput {
         this.mana = mana;
     }
 }
+
+/**
+ * Base class for state metadata outputs.
+ */
+abstract class StateMetadataOutput extends ImmutableFeaturesOutput {
```
Comment (Thoralf-M) [inline bindings/nodejs/lib/types/block/output/output.ts]: Why have this class? AnchorOutput are now the only outputs with a state metadata field, no?

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/block/output/output.ts]: I just kept how it was before with the Account, but I can change it

Comment (Thoralf-M) [reply bindings/nodejs/lib/types/block/output/output.ts]: Would prefer that, don't see a reason to have it separated

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/block/output/output.ts]: Done

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**